sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -111,9 +117,15 @@ export class IdentityClient extends ServiceClient {
     });
 
     try {
-      return await this.sendTokenRequest(webResource, expiresOnParser);
+      const tokenResponse = await this.sendTokenRequest(webResource, expiresOnParser);
+      span.end();
+      return tokenResponse;
     } catch (err) {
-      if (err instanceof AuthenticationError && err.errorResponse.error === "interaction_required") {
+      span.end();
```

Comment:
I'd put span.end() in a finally block rather than in both the try and catch legs.

Reply:
That would work if the `catch` block was not throwing the error again like it does here